src/store/SplitDataProvider.js

```diff
@@ -0,0 +1,362 @@
+              ...split,
+              value: +(equalSplitAmount + 0.01000001).toFixed(2),
+            };
+
+          return { ...split, value: equalSplitAmount };
+        });
+
+        dummyObj = {
+          ...state,
+          splits: newSplits,
+          totalAmount: action.amount,
+          splitsTotalAmount: action.amount,
+        };
+      }
+      return dummyObj;
+    }
+    console.log("you still made it");
+
+    // return {
+    //   ...state,
+    //   totalAmount: updatedAmount,
+    //   splitsPerPerson:
+    //     state.splits.length < 1
+    //       ? "$0"
+    //       : (updatedAmount / state.splits.length).toFixed(2),
+    //   splitsTotalAmount: state.splitType === EQUALLY ? action.amount : 0,
+    // };
+  }
+
+  if (action.type === "EXACT_INPUT") {
+    let dummyObj;
+    const inputAmt = action.amount;
+    const newSplitsWithValues = state.splits.map((split, i) => {
+      return {
+        ...split,
+        value: action.splitId === split.id ? inputAmt : split.value,
+      };
+    });
+    console.log(newSplitsWithValues);
+
+    return {
+      ...state,
+      splits: newSplitsWithValues,
+      splitsTotalAmount: getSplitsTotal(newSplitsWithValues),
+    };
+  }
+  if (action.type === "NAME_CHANGE") {
+    let dummyObj;
+    const inputName = action.name;
+    const newSplitsWithNames = state.splits.map((split, i) => {
+      return {
+        ...split,
+        name: action.splitId === split.id ? inputName : split.name,
+      };
+    });
+
+    return {
+      ...state,
+      splits: newSplitsWithNames,
+    };
+  }
+
+  if (action.type === "ADD") {
+    let dummyObj; // is undefined === console.log(dummyObj);
+    let newSplits;
+
+    const updatedSplits = state.splits.concat(newSplit);
+
+    if (action.splitType === EQUALLY) {
+      if (state.totalAmount === null)
+        return { ...state, splits: updatedSplits };
+
+      if (state.totalAmount != null) {
+        const totalAmountInCents = state.totalAmount * 100;
+        const splitsWithExtraPenny = totalAmountInCents % updatedSplits.length;
+        const equalSplitAmount = +(
+          state.totalAmount / updatedSplits.length
+        ).toFixed(2);
+
+        newSplits = updatedSplits.map((split, i) => {
+          if (i + 1 <= splitsWithExtraPenny)
+            return {
+              ...split,
+              value: +(equalSplitAmount + 0.01000001).toFixed(2),
+            };
+
+          return { ...split, value: equalSplitAmount };
+        });
+        // console.log("updatedSplits", updatedSplits);
+        // console.log("modulo", splitsWithExtraPenny);
+        // console.log(newSplits);
+      }
+      dummyObj = {
+        ...state,
+        splits: newSplits,
+      };
+      return dummyObj;
+    }
+  }
+
+  if (action.type === "REMOVE") {
+    let dummyObj;
+    let newSplits;
+    const updatedSplits = state.splits.filter(
+      (split) => split.id !== action.id
+    );
+    const updatedAmtOfPeople = updatedSplits.length;
```

Comment:
I would suggest almost always avoid using abbreviations and/or acronyms in variable names. There are few times where it won't do harm, but because abbreviations can vary between developers, i.e. not everyone spells abbreviations the same way, it's more prone to human errors. Instead, if use the full spelling of each word, although it's more verbose, it's less prone to human errors and confusion since correct spelling is objective for every developer.
```suggestion
    const updatedAmountOfPeople = updatedSplits.length;
```